csrc/xpu/cutlass_kernels/grouped_gemm_kernel.cpp

```diff
@@ -0,0 +1,471 @@
+      double gflops = options.gflops(cute_average_time / 1000.0,
+                                     options.problem_sizes_host);
+      std::cout << "  Avg runtime : " << cute_average_time << " ms"
+                << std::endl;
+      std::cout << "  GFLOPS      : " << gflops << std::endl;
+    }
+    stream.throw_asynchronous();
+    release();
+    return cutlass::Status::kSuccess;
+  }
+};
+
+void kernel_functor(sycl::queue& stream, void* ptr_A, void* ptr_B, void* ptr_D,
+                    void* ptr_alpha, void* ptr_beta, void* offset, int64_t N,
+                    int64_t K, int64_t groups) {
+  //
+  // Run examples
+  //
+  auto offset_ptr = reinterpret_cast<int64_t*>(offset);
+  Options options(offset_ptr, N, K, groups);
+  // The KernelHardwareInfo struct holds the number of EUs on the GPU with a
+  // given device ID. This information is used by the underlying kernel.
+  cutlass::KernelHardwareInfo hw_info;
+
+  // Change device_id to another value if you are running on a machine with
+  // multiple GPUs and wish to use a GPU other than that with device ID 0.
+  hw_info.sm_count =
+      cutlass::KernelHardwareInfo::query_device_multiprocessor_count(
+          hw_info.device_id);
+
+  using ElementAccumulator = float;
+  using ElementComputeEpilogue = float;
+  using ElementA = cutlass::bfloat16_t;
+  using ElementB = cutlass::bfloat16_t;
+  using ElementOutput = bfloat16_t;
+  using ElementScale = cutlass::bfloat16_t;
+
+  using LayoutA = cutlass::layout::RowMajor;
+  using LayoutB = cutlass::layout::RowMajor;
+  using LayoutC = cutlass::layout::RowMajor;
+  using LayoutD = cutlass::layout::RowMajor;
+
+  using TileShape = Shape<_256, _256, _32>;
+  using GmemTiledCopyA =
+      XE_2D_U16x32x32_LD_N;  // Note: This shape has to match the shape used for
+                             // the scaling factors
+  using GmemTiledCopyB =
+      XE_2D_U16x32x32_LD_V;  // Note: This shape has to match the shape used for
+                             // the scaling factors
+
+  using TiledMma =
+      TiledMMA<MMA_Atom<XE_8x16x16_F32BF16BF16F32_TT>,
+               Layout<Shape<_8, _4, _1>, Stride<_4, _1, _0>>,
+               Tile<Layout<Shape<_8, _8, _4>, Stride<_1, _32, _8>>,
+                    Layout<Shape<_16, _4, _4>, Stride<_1, _64, _16>>, _32>>;
+
+  constexpr int PipelineStages = 2;
+  using GEMMDispatchPolicy =
+      cutlass::gemm::MainloopIntelXeXMX16Group<PipelineStages>;
+  using EpilogueDispatchPolicy = cutlass::epilogue::IntelXeXMX16Group;
+  using EpilogueOp =
+      cutlass::epilogue::fusion::LinearCombination<float_t, float_t>;
+
+  using CollectiveEpilogue =
+      typename cutlass::epilogue::collective::CollectiveBuilder<
+          cutlass::arch::IntelXe, cutlass::arch::OpClassTensorOp, TileShape,
+          Shape<_1, _1, _1>, cutlass::epilogue::collective::EpilogueTileAuto,
+          float, float, float, LayoutC, 1, ElementOutput, LayoutC, 1,
+          EpilogueDispatchPolicy, EpilogueOp>::CollectiveOp;
```

Comment:
Based on `xe_builder.cpp` & this code, it seems you used https://github.com/intel/cutlass-sycl/pull/505 as reference. 
Currently, it's using `EpilogueBuilder`, but I'll replace that code with `CollectiveEpilogue`, which is more configurable.

Thanks!

Reply:
I explicitly attributed the reference in the description of https://github.com/intel/cutlass-sycl/pull/505.
Not only does it give credit to the original author, but it makes maintenance easier.